### ISSUE TITLE
Support MathJax

### ISSUE THREAD

**Issue by 0x6b** [0x6b/copy-selection-as-markdown#8]
From https://github.com/0x6b/copy-selection-as-markdown/issues/1#issuecomment-408766519.

**Comment by 0x6b** [comment]
@allenyllee, I have implemented preliminary MathJax support at  commit 2737095de5faca8433cc55d0d2338acb6fa6cd4a. Since I'm not a MathJax guy, could you please test [this beta](https://github.com/0x6b/copy-selection-as-markdown/blob/2737095de5faca8433cc55d0d2338acb6fa6cd4a/dist/web-ext-artifacts/copy_selection_as_markdown-0.4.0-an%2Bfx.xpi) on your end? I only tested with [this page](https://stats.stackexchange.com/questions/235528/backpropagation-with-softmax-cross-entropy) which you pointed out.

If you have any issue please let me know the URL with exact selection (screenshots would be appreciated as this is important for debugging). Thanks.

**Comment by 0x6b** [comment]
[new beta](https://github.com/0x6b/copy-selection-as-markdown/blob/mathjax/dist/web-ext-artifacts/copy_selection_as_markdown-0.4.1-an%2Bfx.xpi) (0.4.1) is available.

**Comment by allenyllee** [comment]
@0x6b Awesome! very thanks! I think this is the first addon on firefox to support copy as markdown for MathJax. But I still find some bugs, here is the list:

> HackMD is an online editor which support Markdown and Mathjax. I have created a locked page to paste the origin and expected results, you can not modify it, but can create a new one if you like.
> https://hackmd.io/5UH_EAtTTv-QA7N-n4qIcA?both

## issue1:
[derivative - Backpropagation with Softmax / Cross Entropy - Cross Validated](https://stats.stackexchange.com/questions/235528/backpropagation-with-softmax-cross-entropy)
### unexpected line break
![snipaste_2018-08-02_12-33-32](https://user-images.githubusercontent.com/3991134/43562820-798382fe-9651-11e8-859e-74504eba666d.png)
![snipaste_2018-08-02_12-34-42](https://user-images.githubusercontent.com/3991134/43562876-daee0816-9651-11e8-9f78-3114029d9e03.png)
### expected
![snipaste_2018-08-02_12-33-56](https://user-images.githubusercontent.com/3991134/43562837-94af18ea-9651-11e8-8273-b9a11bcd5e1f.png)
![snipaste_2018-08-02_12-35-27](https://user-images.githubusercontent.com/3991134/43562890-e54e73a4-9651-11e8-84cf-a4c3fb21b04e.png)

## issue2:
[LDA(Latent Dirichlet Allocation)主题模型 - CSDN博客](https://blog.csdn.net/aws3217150/article/details/53840029)
### unexpected messed list
![snipaste_2018-08-02_12-35-54](https://user-images.githubusercontent.com/3991134/43562912-09e509ee-9652-11e8-8eda-d3aa4b111124.png)
### expected
![snipaste_2018-08-02_12-36-29](https://user-images.githubusercontent.com/3991134/43562920-11fce6d8-9652-11e8-860b-1898074c1bde.png)

## issue3:
[卷积神经网络(CNN)在句子建模上的应用 | Jey Zhang](http://www.jeyzhang.com/cnn-apply-on-modelling-sentence.html)
### unexpected space
![snipaste_2018-08-02_12-37-18](https://user-images.githubusercontent.com/3991134/43562948-3fac84b2-9652-11e8-9f3a-e7442bcb093d.png)
### expected
![snipaste_2018-08-02_12-51-32](https://user-images.githubusercontent.com/3991134/43563070-d7887a8e-9652-11e8-9c85-5fa30f89ec32.png)

## issue4
[LDA(Latent Dirichlet Allocation)主题模型 - CSDN博客](https://blog.csdn.net/aws3217150/article/details/53840029)
### unexpected messed list(multi-level)
![snipaste_2018-08-02_12-38-45](https://user-images.githubusercontent.com/3991134/43562978-5f73ed76-9652-11e8-8682-728d9f9032ad.png)
### expected
![snipaste_2018-08-02_12-39-12](https://user-images.githubusercontent.com/3991134/43562982-687e215c-9652-11e8-9490-1ba756b08e13.png)


## issue5
[LDA(Latent Dirichlet Allocation)主题模型 - CSDN博客](https://blog.csdn.net/aws3217150/article/details/53840029)
### unexpected: these lines can not be copied
![](https://screenshotscdn.firefoxusercontent.com/images/82477276-5fb0-4a28-bbe7-7c68a71fad45.png)

### expected
> # LDA数学分析
> 
> 上一小节，我们忽略了一个问题，就是如何确定主题和词汇分布，还有文档与词汇的分布，但是要搞明白这个问题，就避免不了一些数学分析了。再次强烈推荐rickjin的《[LDA数学八卦](http://www.52nlp.cn/author/rickjin)》还有Gregor Heinrich有著名的《[Parameter estimation for text analysis](https://users.soe.ucsc.edu/~amichelo/docs/text-est2.pdf)》。此处我只做一些关键扼要的理论推导 :-)
> 
> ## 多项式分布

**Comment by 0x6b** [comment]
@allenyllee much appreciate your efforts! Let me see.

**Comment by 0x6b** [comment]
issue2 seems to be difficult to fix with current addon implementation due to unexpected and ugly DOM structure. Still finding a way to mitigate the situation.

**Comment by allenyllee** [comment]
Hi, could you merge this version (0.4.1) into current version (0.5.2)? Or I have to install these two versions, one for mathjax and one for update. Thanks!

**Comment by 0x6b** [comment]
Alright @allenyllee. I will release this initial mathjax support as new release (possibly opt-in feature). Give me some time as I'm extreme busy recently. Thanks for your patience!

**Comment by 0x6b** [comment]
Hi @allenyllee, thank you for your patience. I've released [v0.6.0](https://addons.mozilla.org/en-US/firefox/addon/copy-selection-as-markdown/versions/0.6.0). Grab the update from your Add-on manager, check **Enable experimental MathJax support**, then hit **Save** to enable this. Thanks!

**Comment by allenyllee** [comment]
Finally, it is very handy for me, very thanks!
